packages/compartment-mapper/src/import-hook.js

```diff
@@ -692,11 +700,17 @@ export function makeImportNowHookMaker(
           compartmentDescriptors,
           compartments,
           absoluteModuleSpecifier: moduleSpecifier,
-          packageLocation,
         });
         if (record) {
           return record;
         }
+
+        // make the module specifier relative to location, since we cannot be sure
+        // any of the subsequent code will accept absolute paths without protest
+        moduleSpecifier = relativeSpecifier(
+          new URL(moduleSpecifier, 'file:').href,
+          compartmentDescriptor.location,
+        );
```

Comment:
While the current implementation of relativeSpecifier makes sense if we've already found a matching compartment and just need to remove the part of the path, it doesn't make sense when we don't know whether the compartment matches.

```js

console.log(relativeSpecifier(
  new URL('/a/b/c.js', 'file:').href,
  '/something/else',
))
```
```
./file:///a/b/c.js
```

Reply:
OK, so then we need to leave as-is.